File: BackEnd/models/task.py
from sqlalchemy import Column, Integer, String, ForeignKey, Boolean, DateTime
from database import Base
from sqlalchemy.orm import relationship

class Task(Base):
    __tablename__ = "tasks"
    id = Column(Integer, primary_key=True, index=True, autoincrement=True)
    title = Column(String(255), index=True, nullable=False)
    isDone = Column(Boolean, nullable=False)
    priority = Column(Integer, nullable=False)
    due_date = Column(DateTime, nullable=True)
    user_id = Column(String(255), ForeignKey("users.username"))
    user = relationship("User", back_populates="tasks")

File: frontend/lib/models/task.dart
class Task {
  final int id;
  final String title;
  final bool isDone;
  final int priority;
  final String? dueDate;
  final String user_id;

  Task({
    required this.id,
    required this.title,
    required this.isDone,
    required this.priority,
    this.dueDate,
    required this.user_id
  });

  factory Task.fromJson(Map<String, dynamic> json) {
    return Task(
      id: json['id'],
      title: json['title'],
      isDone: json['isDone'],
      priority: json['priority'],
      dueDate: json['due_date'],
      user_id: json["user_id"],
    );
  }
}
